pandas/io/stata.py

```diff
@@ -1345,6 +1358,307 @@ def data(self, convert_dates=True, convert_categoricals=True, index=None,
 
         return data
 
+
+    def get_chunk(self, chunksize, convert_dates=True,
```

Comment:
The `read` method should just be a special case of this, at least when `chunksize=None`.  This would allow for removal of redundant code.  After this change, `read` would be ...

```
def read(self):
    return self.get_chunk(None,...)
```

And the output in this case should be identical to the existing version.